platform/security/handler/security-handler-api/src/main/java/org/codice/ddf/security/handler/api/BaseAuthenticationTokenFactory.java

```diff
@@ -0,0 +1,225 @@
+import java.util.List;
+import javax.xml.bind.JAXBElement;
+import org.apache.cxf.helpers.DOMUtils;
+import org.apache.cxf.sts.QNameConstants;
+import org.apache.cxf.ws.security.sts.provider.model.ObjectFactory;
+import org.apache.cxf.ws.security.sts.provider.model.secext.AttributedString;
+import org.apache.cxf.ws.security.sts.provider.model.secext.BinarySecurityTokenType;
+import org.apache.cxf.ws.security.sts.provider.model.secext.PasswordString;
+import org.apache.cxf.ws.security.sts.provider.model.secext.UsernameTokenType;
+import org.apache.wss4j.common.crypto.Merlin;
+import org.apache.wss4j.common.ext.WSSecurityException;
+import org.apache.wss4j.common.token.BinarySecurity;
+import org.apache.wss4j.common.token.X509Security;
+import org.apache.wss4j.dom.WSConstants;
+import org.apache.xml.security.Init;
+import org.codice.ddf.parser.Parser;
+import org.codice.ddf.parser.ParserConfigurator;
+import org.codice.ddf.parser.ParserException;
+import org.codice.ddf.parser.xml.XmlParser;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.w3c.dom.Document;
+import org.w3c.dom.Node;
+
+public class BaseAuthenticationTokenFactory {
+
+  private static final Logger LOGGER =
+      LoggerFactory.getLogger(BaseAuthenticationTokenFactory.class);
+
+  public static final String PKI_TOKEN_ID = "X509PKIPathv1";
+
+  private static final String PKI_TOKEN_VALUE = X509TOKEN_NS + TOKEN_VALUE_SEPARATOR + PKI_TOKEN_ID;
+
+  private Parser parser = new XmlParser();
+
+  private Merlin merlin;
+
+  private String signaturePropertiesPath;
+
+  /** Initializes Merlin crypto object. */
+  public void init() {
+    try {
+      merlin =
+          new Merlin(
+              PropertiesLoader.loadProperties(signaturePropertiesPath),
+              BaseAuthenticationTokenFactory.class.getClassLoader(),
+              null);
+    } catch (WSSecurityException | IOException e) {
+      LOGGER.warn("Unable to read merlin properties file. Unable to validate certificates.", e);
+    }
+    Init.init();
+  }
+
+  /**
+   * Creates a {@link BaseAuthenticationToken} from a given username and password. Uses a {@link
+   * UsernameTokenType} internally to store the username and password.
+   *
+   * @param username - user's username
+   * @param password - user's password
+   * @return a BaseAuthenticationToken containing the given username and password
+   */
+  public BaseAuthenticationToken getTokenFromUsernamePassword(String username, String password) {
+    UsernameTokenType usernameTokenType = new UsernameTokenType();
+    AttributedString user = new AttributedString();
+    user.setValue(username);
+    usernameTokenType.setUsername(user);
+    String usernameToken = null;
+
+    // Add a password
+    PasswordString pass = new PasswordString();
+    pass.setValue(password);
+    pass.setType(WSConstants.PASSWORD_TEXT);
+    JAXBElement<PasswordString> passwordType =
+        new JAXBElement<>(QNameConstants.PASSWORD, PasswordString.class, pass);
```

Comment:
That's something we did in the [WssBasicAuthenticationHandler](https://github.com/codice/ddf/pull/4620/files#diff-582c5f4cb53518062d132f139974ed47L58). Now that we no longer have the UPAuthenticationToken, we need to do this to store the username and password so that the shiro UsernameTokenValidator can validate it correctly.